text/projects/Lukasz_Selwa/Galil_Seiferas.tex

```diff
@@ -0,0 +1,308 @@
+\documentclass{article}
+\usepackage[utf8]{inputenc}
+\usepackage{polski}
+\usepackage[utf8]{inputenc}
+\usepackage[polish]{babel}
+\usepackage[]{amsthm} %lets us use \begin{proof}
+\usepackage[]{amssymb} %gives us the character \varnothing
+\usepackage{algorithm}
+\usepackage[noend]{algorithmic}
+\usepackage{amsmath}
+\usepackage{tikz}
+
+\newtheorem{definition}{Definicja}[section]
+\newtheorem{lemma}{Lemat}[section]
+\newtheorem{theorem}{Twierdzenie}[section]
+
+\DeclareMathOperator{\HRP}{HRP}
+
+\title{Wyszukiwanie wzorca w tekście w stałej pamięci\\ i czasie liniowym}
+\author{Łukasz Selwa}
+\date{Maj 2022}
+
+\begin{document}
+
+\maketitle
+
+\section{Wprowadzenie}
+Naiwny algorytm wyszukiwania wzorca $w$ w tekście $t$ wymaga jedynie stałej pamięci ale zajmuje czas $O(|w| \cdot |t|)$. 
+Znane są algorytmy rozwiązujące problem wyszukiwania wzorca w czasie liniowym $O(|w| + |t|)$, jak algorytm Knutha-Morrisa-Pratta, jednak najczęściej potrzebują liniowej dodatkowej pamięci.
+
+
+W pracy ,,Time-Space-Optimal String Matching'' Z. Galil i J. Seiferas jako pierwsi przedstawili algorytm do wyszukiwania wzorca działający w czasie liniowym i stałej pamięci.
+Niniejszy tekst jest oparty na późniejszej pracy M. Crochemore i W. Rytter ,,Squares, Cubes, and Time-Space Efficient StringS earching'', która zawiera uproszczony opis i dowód oryginalnego algorytmu Galil-Seiferas.
```

Comment:
1. Galila-Seiferasa (everywhere in the document)
2. StringS earching -> String Searching

Reply:
OK, resolved